frontend/packages/ceph-storage-plugin/src/components/ocs-storage-class-form.tsx

```diff
@@ -0,0 +1,174 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { cephBlockPoolResource } from '../constants/resources';
+import { Dropdown } from '@patternfly/react-core/dist/js/components/Dropdown/Dropdown';
+import { DropdownItem } from '@patternfly/react-core/dist/js/components/Dropdown/DropdownItem';
+import { DropdownToggle } from '@patternfly/react-core/dist/js/components/Dropdown';
+import { CaretDownIcon } from '@patternfly/react-icons';
+
+const PoolResourceComponent: React.FC<PoolParams> = ({ onParamChange }) => {
+  const [data, loaded, loadError] = useK8sWatchResource<K8sResourceKind[]>(cephBlockPoolResource);
+  const [isOpen, setOpen] = React.useState<boolean>(false);
+  const [poolName, setPoolName] = React.useState('');
+
+  const handleDropdownChange = (e) => {
+    setPoolName(e.currentTarget.id);
+    onParamChange(e.currentTarget.id);
+  };
+
+  const poolDropdownItems = _.map(data, (pool, key) => (
+    <DropdownItem
+      key={key}
+      component="button"
+      id={pool?.metadata?.name}
+      onClick={(e) => handleDropdownChange(e)}
+    >
+      {pool?.metadata?.name}
+    </DropdownItem>
+  ));
+
+  const onToggle = () => {
+    setOpen(!isOpen);
+  };
+  const onSelect = React.useCallback(() => {
+    setOpen(false);
+  }, [setOpen]);
+
+  return (
+    <>
+      {loaded && !loadError && (
+        <div className="form-group">
+          <label className="co-required" htmlFor="ocs-storage-pool">
+            Pool
+          </label>
+          <Dropdown
+            title="Select Pool"
+            className="dropdown dropdown--full-width"
+            toggle={
+              <DropdownToggle id="toggle-id" onToggle={onToggle} iconComponent={CaretDownIcon}>
+                {poolName || 'Select a Pool'}
+              </DropdownToggle>
+            }
+            isOpen={isOpen}
+            dropdownItems={poolDropdownItems}
+            onSelect={onSelect}
+            id="ocs-storage-pool"
+          />
+          <span className="help-block">Ceph pool into which volume data shall be stored</span>
+        </div>
+      )}
+    </>
+  );
+};
+
+type PoolParams = {
+  onParamChange: Function;
```

Comment:
```suggestion
  onParamChange: (id: string) => void;
```

Reply:
I don't think it makes much of a difference.

Reply:
It does here as `(id: string)` or `(id: string, name:string)` for eg. will both fit.